File: app/src/main/java/com/ayatkhraisat/mvp_example/main/MainContract.java
package com.ayatkhraisat.mvp_example.main;

import com.ayatkhraisat.mvp_example.base.BaseContract;

/**
 * Created by Ayat khraisat  on 12/19/18
 * akhraisat@blessedtreeit.com
 * <p>
 * Project Name: PocMvp
 * <p>
 * Blessed Tree IT
 */
public interface MainContract {

    interface MainPresenter extends BaseContract.BasePresenter {


    }


    interface MainView extends BaseContract.BaseView {

        void showTasksFragment();

        void showSecondFragment(int test);

        void closeNavigationDrawer();
    }
}

File: app/src/main/java/com/ayatkhraisat/mvp_example/main/MainContract.kt
package com.ayatkhraisat.mvp_example.main

import androidx.paging.PagedList
import com.ayatkhraisat.mvp_example.base.BaseContract
import com.ayatkhraisat.mvp_example.models.Model

/**
 * Created by Ayat khraisat  on 1/31/19
 * akhraisat@blessedtreeit.com
 * <p>
 * Project Name: MVP_Architecture
 * <p>
 * Blessed Tree IT
 */
interface MainContract {


    interface MainView  {
        fun showMoviesList(list : PagedList<Model.MovieItem>)
        fun showMoviesDetails()
    }

    interface MainActions {
        fun openMovieDetails();
    }
}